### ISSUE TITLE
inconsistent-mro , but Python does not raise error

### ISSUE THREAD

**Issue by pierluca** [PyCQA/pylint#1377]
### Steps to reproduce
1. `pip install autobahn[twisted]`
2. Try linting the following code:
```
from twisted.internet.protocol import ReconnectingClientFactory
from autobahn.twisted import websocket


class RetryFactory(websocket.WampWebSocketClientFactory,
                   ReconnectingClientFactory):
    pass
```

### Current behavior
```
poc.py:5: [R0901(too-many-ancestors), RetryFactory] Too many ancestors (11/7)
poc.py:5: [E0240(inconsistent-mro), RetryFactory] Inconsistent method resolution order for class 'RetryFactory'
```

### Expected behavior
```
poc.py:5: [R0901(too-many-ancestors), RetryFactory] Too many ancestors (11/7)
```
Too many ancestors is expected, given the library imports at hand.
However, we can't reproduce the inconsistent method resolution order on a Python execution of the script. Isn't this a false positive?

### pylint --version output
```
pylint 1.6.5, 
astroid 1.4.9
Python 2.7.9 (default, Jun 29 2016, 13:08:31) 
[GCC 4.9.2]
```

**Comment by PCManticore** [comment]
It might be a false positive. We are trying to compute the MRO, but we might have some edge cases we didn't handle properly when we first implemented the algorithm. Thank you, will take a look.

**Comment by rogalski** [comment]
There is a magic decorator in twisted source that converts old-style class to new-style classes. Most likely Pylint still sees this class as old-style class, which changes the way MRO works.

**Comment by rogalski** [comment]
```
class Factory(object):  # make it old-style class to trigger an error
    pass


class ClientFactory(Factory):
    pass


class ReconnectingClientFactory(ClientFactory):
    pass


class WebSocketAdapterFactory(object):
    pass


class WebSocketClientFactory(WebSocketAdapterFactory, ClientFactory):
    pass


class WampWebSocketClientFactory(WebSocketClientFactory):
    pass


class RetryFactory(WampWebSocketClientFactory, ReconnectingClientFactory):
    pass
```

Slightly smaller, standalone reproduction.

**Comment by PCManticore** [comment]
Closed in astroid's master by https://github.com/PyCQA/astroid/commit/5092504bd476fa24a95bf9e818f4523f9b67f4c5
